src/resources/2017/Robortender.ts

```diff
@@ -0,0 +1,55 @@
+import { haveFamiliar, Item, Monster, Phylum } from "kolmafia";
+import { get } from "../../property";
+import { $familiar, $item, $phylum } from "../../template-string";
+
+/**
+ * The Robortender itself
+ */
+export const familiar = $familiar`Robortender`;
+
+/**
+ * @returns Whether you have the Robortender in your terrarium/available
+ */
+export function have(): boolean {
+  return haveFamiliar(familiar);
+}
+
+const phylumDrops = new Map<Phylum, Item>([
+  [$phylum`Bug`, $item`pickled grasshopper`],
+  [$phylum`Constellation`, $item`bottle of anís`],
+  [$phylum`Demon`, $item`bottle of novelty hot sauce`],
+  [$phylum`Elemental`, $item`elemental sugarcube`],
+  [$phylum`Elf`, $item`peppermint sprig`],
+  [$phylum`Fish`, $item`bottle of clam juice`],
+  [$phylum`Goblin`, $item`cocktail mushroom`],
+  [$phylum`Hippy`, $item`shot of granola liqueur`],
+  [$phylum`Hobo`, $item`can of cherry-flavored sterno`],
+  [$phylum`Horror`, $item`lump of black ichor`],
+  [$phylum`Humanoid`, $item`bottle of gregnadigne`],
+  [$phylum`Mer-kin`, $item`bottle of Crème de Fugu`],
+  [$phylum`Orc`, $item`baby oil shooter`],
+  [$phylum`Penguin`, $item`fish head`],
+  [$phylum`Pirate`, $item`limepatch`],
+  [$phylum`Plant`, $item`pile of dirt`],
+  [$phylum`Slime`, $item`slime shooter`],
+  [$phylum`Weird`, $item`imaginary lemon`],
+]);
+
+/**
+ *
+ * @param target The phylum or monster you want to know the robortender drop of
+ * @returns The robortender drop associated with that phylum or monster
+ */
+export function dropFrom(target: Monster | Phylum): Item {
+  const phylum = target instanceof Monster ? target.phylum : target;
+  return phylumDrops.get(phylum) ?? $item`none`;
+}
+
+/**
+ * Determines the probability of getting a robortender drop based on number of drops received
+ * @param dropNumber The number of drops to assume you've already received; defaults to mafia's tracked amount
+ * @returns The probability of getting a robort drop
+ */
+export function dropChance(dropNumber = get("_roboDrops")): number {
```

Comment:
do we need to push the necessary mafia version for libram to ensure we've got the correct _roboDrops? i vaguely recall there being some small errors with the initial implementation. i suppose this isn't necessarily that big of an issue since this technically just floors at 20%, but might as well bring it up i guess.

Reply:
iirc libram doesn't have an associated mafia version, just the scripts that call it.

we probably should have one, though!

Reply:
I think that can be handled scriptside as needed. Though to a similar point, does new libram version automatically update property typing? otherwise that won't work, because get() will return a string for that prop, since it doesn't know what it is.